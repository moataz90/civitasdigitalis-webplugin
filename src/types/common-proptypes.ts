export interface IPropsChildable {
	children?: any;
}

export interface IPropsStyledComponents {
	className?: string;
}