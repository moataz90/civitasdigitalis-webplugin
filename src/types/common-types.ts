export type MessageID = string;
export type Timestamp = number;

export enum WSMessageEvent {
	UserNotFound = 'user_not_found',
	UserFound = 'user_found',
	UserCreated = 'user_created',
	PUIDInvalid = 'puid_invalid',
	PageIDNotFound = 'pageid_not_found',
	PageIDMissing = 'pageid_missing',
	VerfiyTokenMissing = 'verifytoken_missing',
	ConnectionRejected = 'connection_rejected',
	NewMessage = 'new_message',
	NewAction = 'new_action'
}

export interface IWebsocketMessageEvent<PayloadTyp = string> {
	statusCode: number;
	connected: boolean;
	type: WSMessageEvent;
	payload?: PayloadTyp;
}