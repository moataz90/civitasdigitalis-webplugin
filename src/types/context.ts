import { IConfig } from './../config/config-schema';

export interface IContext {
	config: IConfig;
}