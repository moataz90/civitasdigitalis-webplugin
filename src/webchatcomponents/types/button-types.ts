export enum MessageButtonType {
	Url,
	Postback,
	Call,
	Share
}

export interface IMessagePayloadButton {
	type: MessageButtonType;
	title: string;
	payload?: string;
}

export type ButtonCallback = (type: number, title: string, payload: any) => void;