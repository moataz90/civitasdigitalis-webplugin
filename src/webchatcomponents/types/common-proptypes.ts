import * as React from 'react';

export interface IPropsChildable {
	children?: any;
}

export interface IPropsStyledComponents {
	className?: string;
}

export interface IPropsCustomStyleable {
	css?: React.CSSProperties;
}