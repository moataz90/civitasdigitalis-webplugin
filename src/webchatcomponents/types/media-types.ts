export enum MessageMediaType {
	Audio,
	Video,
	Image,
	File
}