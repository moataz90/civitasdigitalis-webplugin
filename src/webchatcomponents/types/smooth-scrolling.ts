export type EasingFunction = (t: number) => number;

export type ScrollSmothCallback = (interpolatedValue: number) => void;

export type SmoothScrolling = (
	startValue: number,
	targetValue: number,
	interpolation: EasingFunction,
	animationTime: number,
	callbackUpdate: ScrollSmothCallback,
	callbackFinish?: () => void
) => void;