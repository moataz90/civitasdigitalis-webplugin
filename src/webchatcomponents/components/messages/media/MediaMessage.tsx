import * as React from 'react';
import { MessageMediaType } from '../../../types/media-types';
import { ImageMessage } from './ImageMessage';
import { IMessageBaseProps } from '../Message';

export interface IMediaMessageProps extends IMessageBaseProps {
	mediaUrl: string;
	mediaType: MessageMediaType;
	maxWidth: string | number;
	onLoadMedia: (success: boolean, width: number, height: number) => void;
}

export const MediaMessage: React.StatelessComponent<IMediaMessageProps> =
	({ mediaUrl, mediaType, maxWidth, isOwnMessage, isLastMessage, onLoadMedia }) => {
		switch (mediaType) {
			case MessageMediaType.Image:
				return (
					<ImageMessage
						imageUrl={mediaUrl}
						maxWidth={maxWidth}
						isOwnMessage={isOwnMessage}
						isLastMessage={isLastMessage}
						onLoadImage={onLoadMedia}
					/>
				);

			default:
				return null;

		}
	};