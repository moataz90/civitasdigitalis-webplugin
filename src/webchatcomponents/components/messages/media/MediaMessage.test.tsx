import * as React from 'react';
import * as renderer from 'react-test-renderer';
import 'jest-styled-components';
import { MediaMessage } from './MediaMessage';
import { MessageMediaType } from '../../../types/media-types';

// The following test cases are based on jest snapshot testing!
// For more information on how snapshot tests are working, see
// https://jest-bot.github.io/jest/docs/snapshot-testing.html

test('render media message of type image', () => {
	const domTree = renderer.create(
		<MediaMessage
			mediaType={MessageMediaType.Image}
			mediaUrl="http://wiesmann.codiferes.net/share/bitmaps/test_pattern.svg"
			maxWidth={200}
			onLoadMedia={() => undefined}
			isOwnMessage={true}
			isLastMessage={false}
		/>
	).toJSON();

	expect(domTree).toMatchSnapshot();
});

test('render media message of other type', () => {
	const domTree = renderer.create(
		<MediaMessage
			mediaType={MessageMediaType.Video}
			mediaUrl="http://wiesmann.codiferes.net/share/bitmaps/test_pattern.svg"
			maxWidth={200}
			onLoadMedia={() => undefined}
			isOwnMessage={true}
			isLastMessage={false}
		/>
	).toJSON();

	expect(domTree).toBe(null);
});