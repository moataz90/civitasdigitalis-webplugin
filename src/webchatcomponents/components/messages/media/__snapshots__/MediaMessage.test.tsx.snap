// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render media message of type image 1`] = `
.c0.c0.c0 {
  width: 100%;
  position: relative;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding-right: 10px;
}

.c1.c1.c1 {
  width: 100%;
  border-radius: 8px;
}

<div
  className="c0"
  style={
    Object {
      "justifyContent": "flex-end",
    }
  }
>
  <div
    style={
      Object {
        "minHeight": "1px",
        "position": "relative",
        "width": "200px",
      }
    }
  >
    <img
      className="c1"
      src="http://wiesmann.codiferes.net/share/bitmaps/test_pattern.svg"
    />
  </div>
</div>
`;
